app/process_proposal.go

```diff
@@ -85,51 +83,42 @@ func (app *App) ProcessProposal(req abci.RequestProcessProposal) abci.ResponsePr
 			continue
 		}
 
-		for _, msg := range tx.GetMsgs() {
-			if sdk.MsgTypeURL(msg) != types.URLMsgPayForBlob {
-				continue
-			}
+		msgs := tx.GetMsgs()
+		// ignore transactions with more than a single sdk.Msg since we do not
+		// yet support arbitrary sdk.Tx
+		if len(msgs) != 1 {
+			continue
+		}
```

Comment:
If it's not possible for a correct node to generate this in `PrepareProposal` then we should outright reject then entire proposal. Same for line 96

Reply:
good point, I agree and don't see why we shouldn't change this now. e9a8014d39f49cb1b832d3272ca4f5d8ba46bdc9

I tried to test this, but it would take a ton of code to hit this test, as we'd basically have to write a custom share splitting method just to create an other wise valid block.